### PR TITLE
[DPEDE-324] Create Dropdown menu web component in UXPin

### PR DESCRIPTION
**Pull request by soujanreddy**

### CODE REVIEW

File: src/components/DropdownMenuWc/presets/0-default.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import DropdownMenuWc from '../DropdownMenuWc';
+
+export default (
+  <DropdownMenuWc
+    uxpId="dropdown-menu-wc"
+    item1="Item 1"
+    item2="Item 2"
+    item3="Item 3"
+    selectedItem={1}
+    dropdownPosition="bottom-start"
```
Comment (josecarlosgonzalezv) [inline src/components/DropdownMenuWc/presets/0-default.jsx:11]:
```suggestion
```

File: src/components/DropdownMenuWc/DropdownMenuWc.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        items.push({
+          title,
+          iconLeft: props[`iconLeft${i}`],
+          iconRight: props[`iconRight${i}`],
+          description: props[`item${i}Description`],
+          onSelect: props[`select${i}`],
+        });
+      }
+    }
+    return items;
```
Comment (josecarlosgonzalezv) [inline src/components/DropdownMenuWc/DropdownMenuWc.js:28]:
```suggestion
    }
    
    return items;
```

File: src/components/DropdownMenuWc/DropdownMenuWc.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  };
+  // #endregion
+
+  // #region Attributes
+  const items = getDropdownItems(props);
+
+  const hasDescription = items.some((item) => item.description);
```
Comment (josecarlosgonzalezv) [inline src/components/DropdownMenuWc/DropdownMenuWc.js:35]:
```suggestion
  const items = getDropdownItems(props);
  const hasDescription = items.some((item) => item.description);
```

File: src/components/DropdownMenuWc/DropdownMenuWc.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  select6: PropTypes.func,
+  select7: PropTypes.func,
+  select8: PropTypes.func,
+  select9: PropTypes.func,
+  select10: PropTypes.func,
+};
+
+DropdownMenuWc.defaultProps = {
+  mode: 'base',
+  active: true,
```
Comment (josecarlosgonzalezv) [inline src/components/DropdownMenuWc/DropdownMenuWc.js:117]:
```suggestion
```

File: src/components/DropdownMenuWc/DropdownMenuWc.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+  const hasDescription = items.some((item) => item.description);
+  // #endregion
+
+  // TODO: Implement custom width and height for dropdown menu (both with items and when empty) on chi
+  // TODO: Retain selection with active selected items should be implemented on chi
+  // TODO: Visible items should be able to add scroll on dropdown menu and must be compatible with custom height
+  // TODO: Appropriate padding must be provided between item and icon right
+  return (
+    <chi-dropdown
+      active={active}
```
Comment (josecarlosgonzalezv) [inline src/components/DropdownMenuWc/DropdownMenuWc.js:44]:
```suggestion
      active
```

**Review by josecarlosgonzalezv — CHANGES_REQUESTED** [review]

File: src/components/DropdownMenuWc/DropdownMenuWc.js (hunk 5 above)
Comment (soujanreddy) [reply src/components/DropdownMenuWc/DropdownMenuWc.js:44]: @josecarlosgonzalezv We need this active flag to toggle dropdown menu to show and hide to be in sync with current implementation.

**Pull request closed without merging**